geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/IngestRasterCommand.scala

```diff
@@ -38,75 +35,104 @@ class IngestRasterCommand(parent: JCommander) extends Command with AccumuloPrope
     val fmt = Option(params.format).getOrElse(getFileExtension(params.file))
     fmt match {
       case TIFF | DTED =>
-        ingest
+        ingest(params.mode)
       case _         =>
         logger.error("Error: File format not supported for file " + params.file + ". Supported formats " +
           "are geotiff and DTED")
     }
   }
 
-  def ingest() {
-    val cs = createCoverageStore(params)
-
-    val args: Map[String, Option[String]] = Map(
-      IngestRasterParams.FILE_PATH -> Some(params.file),
-      IngestRasterParams.FORMAT -> Some(Option(params.format).getOrElse(getFileExtension(params.file))),
-      IngestRasterParams.TABLE -> Option(params.table),
-      IngestRasterParams.TIME -> Option(params.timeStamp),
-      IngestRasterParams.PARLEVEL -> Option(params.parLevel.toString)
-    )
-
-    val ingester = new SimpleRasterIngest(args, cs)
-    ingester.runIngestTask() match {
-      case Success(info) => logger.info("Ingestion is done.")
-      case Failure(e) => throw new RuntimeException(e)
+  //In both modes, input raster file(s) are local files.
+  //In local mode, file(s) is(are) directly ingested into an Accumulo table.
+  //In remote mode, file(s) is(are) serialized and stored into HDFS as sequence file(s),
+  //and a scalding job is executed to ingest sequence file(s) into an Accumulo table from HDFS.
+  def ingest(mode: String) {
+    val baseRasterIngestParams = getRasterIngestParams
+    mode.toLowerCase match {
+      case "local" =>
+        val simpleIngester =
+          new LocalRasterIngest(baseRasterIngestParams + (IngestRasterParams.FILE_PATH  -> Some(params.file)))
+        simpleIngester.runIngestTask() match {
+          case Success(info) => logger.info("Local ingestion is done.")
+          case Failure(e) => throw new RuntimeException(e)
+        }
+      case "remote" =>
+        val rasterSerializer =
+          new RasterFilesSerialization(baseRasterIngestParams + (IngestRasterParams.FILE_PATH  -> Some(params.file)))
+        rasterSerializer.runSerializationTask() match {
+          case Success(outPath) =>
+            logger.info("Raster files serialization is done.")
+            new RemoteRasterIngest(baseRasterIngestParams + (IngestRasterParams.FILE_PATH  -> Some(outPath))).run
+          case Failure(e) => throw new RuntimeException(e)
+        }
+      case _ => logger.error("Unsupported ingestion mode")
```

Comment:
If a user mistypes something, can we hit this error?  If so, I'd suggest expanding it to hint at usage.